src/util/sharing_node.h

```diff
@@ -12,339 +12,455 @@ Author: Daniel Poetzl
 #ifndef CPROVER_UTIL_SHARING_NODE_H
 #define CPROVER_UTIL_SHARING_NODE_H
 
-#include <list>
+//#define DEBUG
```

Comment:
Can this be removed instead of commented out? I think it can be passed as a CLI compiler supplied symbol during compilation with `-DDEBUG`. Other than that the name is too generic - is the symbol `DEBUG` reserved across the whole of CBMC for debugging support like this, or is this code debugging support just specific to this code? If so, can it be renamed to something else if it can't be deleted?

Reply:
I'll rename it to `SN_DEBUG` (the other macros in the sharing node file also start with `SN`).